[DRAFT] OSDOCS#11879: Updated cluster-admin creation process (#81465)
openshift/openshift-docs: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc

? Domain prefix: <domain_prefix> <1>
? Deploy cluster with Hosted Control Plane (optional): No
? Create cluster admin user: Yes <2>
? Create custom password for cluster admin: No <3>
I: cluster admin user is cluster-admin

> I: cluster admin password is abcd1-eFgh2-jKLm3-NopQ4

Reviewer: See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-reference

I had the security team contact me in the past about using password syntax that represents real passwords.

![Screenshot from 2024-10-02 12-05-26](https://github.com/user-attachments/assets/b1b98a20-aef9-46b3-be53-4ef05c62184a)

Author: Roger that, thankyou!

> ? OpenShift version: 4.17.0 <4>

Reviewer: ```suggestion
? OpenShift version: <openshift_version> <4>
```

I: Using arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Worker-Role for the Worker role
I: Using arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Support-Role for the Support role
? External ID (optional): <7>
? Operator roles prefix: <cluster_name>-<random_string> <8>
? Deploy cluster using pre registered OIDC Configuration ID:
? Tags (optional) <9>
? Multiple availability zones (optional): No <10>
? AWS region: us-east-1
? PrivateLink cluster (optional): No
? Install into an existing VPC (optional): Yes <11>
? Select availability zones (optional): No
? Enable Customer Managed key (optional): No <12>
? Compute nodes instance type (optional):
? Enable autoscaling (optional): No
? Compute nodes: 2
? Additional Security Group IDs (optional): <13>
? > [*]  sg-0e375ff0ec4a6cfa2 ('sg-1')
? > [ ]  sg-0e525ef0ec4b2ada7 ('sg-2')
? Machine CIDR: 10.0.0.0/16
? Service CIDR: 172.30.0.0/16
? Pod CIDR: 10.128.0.0/14
? Host prefix: 23
? Encrypt etcd data (optional): No <14>
? Disable Workload monitoring (optional): No
I: Creating cluster '<cluster_name>'
I: To create this cluster again in the future, you can run:

>    rosa create cluster --cluster-name <cluster_name> --role-arn arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Installer-Role --support-role-arn arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Support-Role --master-iam-role arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-ControlPlane-Role --worker-iam-role arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Worker-Role --operator-roles-prefix <cluster_name>-<random_string> --region us-east-1 --version 4.17.0 --additional-compute-security-group-ids sg-0e375ff0ec4a6cfa2 --additional-infra-security-group-ids sg-0e375ff0ec4a6cfa2 --additional-control-plane-security-group-ids sg-0e375ff0ec4a6cfa2 --replicas 2 --machine-cidr 10.0.0.0/16 --service-cidr 172.30.0.0/16 --pod-cidr 10.128.0.0/14 --host-prefix 23 <15>

Reviewer: Would moving the example to the <15> callout be better?

Author: After integrating some changes in main, it's now `<16>`

I: To view a list of clusters and their status, run 'rosa list clusters'
I: Cluster '<cluster_name>' has been created.
I: Once the cluster is installed you will need to add an Identity Provider before you can login into the cluster. See 'rosa create idp --help' for more information.
...
----
<1> Optional. When creating your cluster, you can customize the subdomain for your cluster on `*.openshiftapps.com` using the `--domain-prefix` flag. The value for this flag must be unique within your organization, cannot be longer than 15 characters, and cannot be changed after cluster creation. If the flag is not supplied, an autogenerated value is created that depends on the length of the cluster name. If the cluster name is fewer than or equal to 15 characters, that name is used for the domain prefix. If the cluster name is longer than 15 characters, the domain prefix is randomly generated to a 15 character string.

> <2> When creating your cluster, you can create a local administrator user (`cluster-admin`) for your cluster. This automatically configures an htpasswd identity provider for the `cluster-admin` user.

Reviewer: ```suggestion
<2> When creating your cluster, you can create a local administrator user (`cluster-admin`) for your cluster. This automatically configures an `htpasswd` identity provider for the `cluster-admin` user.
```

?

> <3> You can create a custom password for the `cluster-admin` user, or have the system generate a password. If you do not create a custom password, the generated password is displayed in the command line output. If you specify a custom password, the password must be at least 14 characters (ASCII-standard) without any whitespace. Once defined, the password is hashed and transported securely.

Reviewer: ```suggestion
<3> You can create a custom password for the `cluster-admin` user, or have the system generate a password. If you do not create a custom password, the generated password is displayed in the command line output. If you specify a custom password, the password must be at least 14 characters (ASCII-standard) without any whitespace. When defined, the password is hashed and transported securely.
```

<4> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.17.0`.

> <5> Optional: Specify 'optional' to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify 'required' to configure all EC2 instances to use IMDSv2 only.

Reviewer: ```suggestion
<5> Optional: Specify `optional` to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify `required` to configure all EC2 instances to use IMDSv2 only.
```

====
endif::openshift-rosa[]
+

> <6> If you have more than one set of account roles for your cluster version in your AWS account, an interactive list of options is provided.

Reviewer: ```suggestion
<6> If you have more than one set of account roles for your cluster version in your {AWS} account, an interactive list of options is provided.
```

<7> Optional: Specify an unique identifier that is passed by {product-title} and the OpenShift installer when an account role is assumed. This option is only required for custom account roles that expect an external ID.

> <8> By default, the cluster-specific Operator role names are prefixed with the cluster name and a random 4-digit hash. You can optionally specify a custom prefix to replace `<cluster_name>-<hash>` in the role names. The prefix is applied when you create the cluster-specific Operator IAM roles. For information about the prefix, see _Defining an Operator IAM role prefix_.

Reviewer: I was expecting "Defining an Operator IAM role prefix" to be in the same document/doc book, but it is not. Maybe at a link in an add. resources section?

Author: Cheers, I've corrected the title and added to the existing additional resources section

+
[NOTE]
====
If you specified custom ARN paths when you created the associated account-wide roles, the custom path is automatically detected. The custom path is applied to the cluster-specific Operator roles when you create them in a later step.
====

> <9> Optional: Specify a tag that is used on all resources created by {product-title} in AWS. Tags can help you manage, identify, organize, search for, and filter resources within AWS. Tags are comma separated, for example: "key value, data input".

Reviewer: ```suggestion
<9> Optional: Specify a tag that is used on all resources created by {product-title} in AWS. Tags can help you manage, identify, organize, search for, and filter resources within AWS. Tags are comma separated, for example: `key value, data input`.
```
